### PR TITLE
Closes #2433: submodules must start with pro.taskana

### LINKED ISSUE #2433: Release Pipeline does not work with new SubModule dmn-xlsx-converter
see https://github.com/Taskana/taskana/actions/runs/6829760214

### PR DESCRIPTION
**Pull request by arolfes**

### CODE REVIEW

File: rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/SpringSecurityToJaasFilter.java [hunk 1]
```diff
@@ -68,7 +68,7 @@ protected Optional<Subject> obtainSubject() {
       return Optional.empty();
     }
     // TODO replace with Subject.current() when migrating to newer Version then 17
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/SpringSecurityToJaasFilter.java:70]:
```suggestion
    // TODO replace with Subject.current() when migrating to newer Version than 17
```

File: rest/taskana-rest-spring-example-boot/src/main/java/pro/taskana/example/boot/security/BootWebSecurityConfigurer.java [hunk 2]
```diff
@@ -123,15 +125,22 @@ protected void addLoginPageConfiguration(HttpSecurity http) throws Exception {
             authorizeHttpRequests -> authorizeHttpRequests.anyRequest().fullyAuthenticated())
         .formLogin(
             formLogin ->
-                formLogin.loginPage("/login").failureUrl("/login?error").defaultSuccessUrl("/"))
+                formLogin
+                    .loginPage("/login")
+                    .failureUrl("/login?error")
+                    .defaultSuccessUrl("/index.html")
+                    .permitAll()
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-example-boot/src/main/java/pro/taskana/example/boot/security/BootWebSecurityConfigurer.java:132]: Sure that `permitAll` is correct here? I am no spring security expert but this sounds weird.

File: rest/taskana-rest-spring-example-boot/src/main/java/pro/taskana/example/boot/security/BootWebSecurityConfigurer.java [hunk 3]
```diff
@@ -123,15 +125,22 @@ protected void addLoginPageConfiguration(HttpSecurity http) throws Exception {
...
         .logout(
             logout ->
                 logout
                     .invalidateHttpSession(true)
                     .clearAuthentication(true)
                     .logoutRequestMatcher(new AntPathRequestMatcher("/logout"))
                     .logoutSuccessUrl("/login?logout")
-                    .deleteCookies("JSESSIONID"));
+                    .deleteCookies("JSESSIONID")
+                    .permitAll()
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-example-boot/src/main/java/pro/taskana/example/boot/security/BootWebSecurityConfigurer.java:142]: Sure that `permitAll` is correct here? I am no spring security expert but this sounds weird.

File: rest/taskana-rest-spring-example-boot/src/main/java/pro/taskana/example/boot/security/BootWebSecurityConfigurer.java (hunk 2 above)
Comment (sfrevel) [reply rest/taskana-rest-spring-example-boot/src/main/java/pro/taskana/example/boot/security/BootWebSecurityConfigurer.java:132]: isn't this just for the login page path?

File: rest/taskana-rest-spring-example-boot/src/main/java/pro/taskana/example/boot/security/BootWebSecurityConfigurer.java (hunk 3 above)
Comment (sfrevel) [reply rest/taskana-rest-spring-example-boot/src/main/java/pro/taskana/example/boot/security/BootWebSecurityConfigurer.java:142]: isn't this just for the login page path?

File: rest/taskana-rest-spring-example-boot/src/main/java/pro/taskana/example/boot/security/BootWebSecurityConfigurer.java (hunk 2 above)
Comment (arolfes) [reply rest/taskana-rest-spring-example-boot/src/main/java/pro/taskana/example/boot/security/BootWebSecurityConfigurer.java:132]: Hello @mustaphazorgati , @sfrevel ,

@sfrevel is absolutely right.

The `permitAll()` specified here only refers to the authentication part. It is a bit confusing that you specify the `defaultSuccessUrl("/index.html")` and you might think that this is also permitted. However, the AuthenticationHandler ensures that this is not the case.

Without the `permitAll()` there is an infinite redirect to the default page `/taskana/index.html` -> which is secured and tries to call the login page. 

Only the login url and the login failure url are allowed here. Nothing else.

See the Spring security code for more details. I did not bother to look for the explanation in the Spring Security docs. 
* `org.springframework.security.config.annotation.web.configurers.FormLoginConfigurer`
* `org.springframework.security.config.annotation.web.configurers.AbstractAuthenticationFilterConfigurer`
* `org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter`
* `org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter`

Alternatively, we could specify the login url and the login failure url as exceptions above. (where the exceptions for CSS and images are defined). **Should I try this?**

I have tested the current config as follows

- user is not logged in
- call `localhost:8080/taskana` -> redirect to login
- do not log in!
- call `localhost:8080/taskana/index.html` -> redirect to login page
- call `localhost:8080/taskana/taskana/something` -> redirect to login page
- call `localhost:8080/taskana/#/taskana/administration/workbaskets` -> redirect to login page

login user now -> everything works as expected

is this enough explanation? or should we have a look at the code together?

File: rest/taskana-rest-spring-example-boot/src/main/java/pro/taskana/example/boot/security/BootWebSecurityConfigurer.java (hunk 3 above)
Comment (arolfes) [reply rest/taskana-rest-spring-example-boot/src/main/java/pro/taskana/example/boot/security/BootWebSecurityConfigurer.java:142]: Hello @mustaphazorgati, @sfrevel 
please see my comment for login section. The same applies here.

File: rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/SpringSecurityToJaasFilter.java (hunk 1 above)
Comment (arolfes) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/SpringSecurityToJaasFilter.java]: I updated all the same comments in the code with your suggestion. :)
Thanks for finding my bad english spelling/writing issues :)

File: rest/taskana-rest-spring-example-boot/src/main/java/pro/taskana/example/boot/security/BootWebSecurityConfigurer.java (hunk 2 above)
Comment (mustaphazorgati) [reply rest/taskana-rest-spring-example-boot/src/main/java/pro/taskana/example/boot/security/BootWebSecurityConfigurer.java:132]: This is fine for me. Thank you for the explanation!

**Review by mustaphazorgati — APPROVED** [review]

**Pull request merged**